File: webapp/src/model/GitHubUser.js

export default class GitHubUser {
  constructor(data){
    this.id = data.id;
    this.login = data.login;
    this.bio = data.bio;
    this.avatar_url = data.avatar_url;
    this.name = data.name;
    this.html_url = data.html_url;
  }

  parseToJson() {
    return {
      id: this.id,
      login: this.login,
      bio: this.bio,
      avatar_url: this.avatar_url,
      name: this.name,
      html_url: this.html_url,
    };
  }
}

File: src/main/java/assignment/eligosoft/github/fetcher/model/GithubUser.java
package assignment.eligosoft.github.fetcher.model;

import javax.persistence.Entity;
import javax.persistence.Id;

@Entity
public class GithubUser {

  @Id
  private Long id;
  private String login;
  private String bio;
  private String avatar_url;
  private String name;
  private String html_url;

  public GithubUser() {
  }

  public Long getId() {
    return id;
  }

  public void setId(Long id) {
    this.id = id;
  }

  public String getLogin() {
    return login;
  }

  public void setLogin(String login) {
    this.login = login;
  }

  public String getBio() {
    return bio;
  }

  public void setBio(String bio) {
    this.bio = bio;
  }

  public String getAvatar_url() {
    return avatar_url;
  }

  public void setAvatar_url(String avatar_url) {
    this.avatar_url = avatar_url;
  }

  public String getName() {
    return name;
  }

  public void setName(String name) {
    this.name = name;
  }

  public String getHtml_url() {
    return html_url;
  }

  public void setHtml_url(String html_url) {
    this.html_url = html_url;
  }
}
